src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -247,6 +241,981 @@ public bool NeedsRefresh()
+                                    }
+                                    else if (reader.TokenType != JsonTokenType.Null)
+                                    {
+                                        throw CreateFeatureFlagFormatException(
+                                            $"{FeatureManagementConstants.PercentileAllocation}[{i}]",
+                                            settingKey,
+                                            reader.TokenType.ToString(),
+                                            JsonTokenType.StartObject.ToString());
+                                    }
+
+                                    i++;
+                                }
+
+                                featureAllocation.Percentile = percentileAllocations;
+                            }
+                            else if (reader.TokenType != JsonTokenType.Null)
+                            {
+                                throw CreateFeatureFlagFormatException(
+                                    FeatureManagementConstants.PercentileAllocation,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    JsonTokenType.StartArray.ToString());
+                            }
+
+                            break;
+                        }
+
+                    case FeatureManagementConstants.Seed:
+                        {
+                            if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                            {
+                                featureAllocation.Seed = reader.GetString();
+                            }
+                            else if (reader.TokenType != JsonTokenType.Null)
+                            {
+                                throw CreateFeatureFlagFormatException(
+                                    FeatureManagementConstants.Seed,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    JsonTokenType.String.ToString());
+                            }
+
+                            break;
+                        }
+
+                    default:
+                        reader.Skip();
+
+                        break;
+                }
+            }
+
+            return featureAllocation;
+        }
+
+        private FeatureUserAllocation ParseFeatureUserAllocation(ref Utf8JsonReader reader, string settingKey)
+        {
+            var featureUserAllocation = new FeatureUserAllocation();
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    continue;
+                }
+
+                string userAllocationPropertyName = reader.GetString();
+
+                switch (userAllocationPropertyName)
```

Comment:
Should we handle the case that one of `variant` and `users` is missing? For example,
``` json
"allocation": {
		"user": [
			{
				"users": [
					"a",
					"b"
				]
			}
		]
	}
```
This user allocation will not take effect. Do we still need to put it in the configuration?

Reply:
Good question, I think it should check to see if they aren't empty, so if there's anything inside of "user", either a list of users or a variant name, then it would be added. I thought since we shouldn't necessarily be doing any feature management logic, it would make sense to just exclude it if it's completely empty, but otherwise add the object. So in this case that you put up, I think it should be added. I don't think we need to throw an exception though, since we never did something like that if `conditions` was empty, for example.

Reply:
Agree with Amer here. Feature Management library is responsible for checking the validity of a feature flag. Provider library just reads and flattens whatever is stored in AppConfig.